Components/Sources/Components/Utility/HTMLUtils.swift

```diff
@@ -0,0 +1,513 @@
+import Foundation
+import SwiftUI
+import UIKit
+
+public struct HtmlUtils {
+    
+    // MARK: - Shared - Nested Types
+    
+    public struct Styles {
+        let font: UIFont
+        let boldFont: UIFont
+        let italicsFont: UIFont
+        let boldItalicsFont: UIFont
+        let color: UIColor
+        let linkColor: UIColor
+    }
+    
+    private enum ListType {
+        case ordered
+        case unordered
+    }
+    
+    private struct StyleData {
+        var openNSRanges: [NSRange] = []
+        var completeNSRanges: [NSRange] = []
+        var targetAttributeValues: [String] = []
+    }
+    
+    private struct AllStyleData {
+        let bold: StyleData
+        let italics: StyleData
+        let link: StyleData
+        let `subscript`: StyleData
+        let superscript: StyleData
+        let strikethorugh: StyleData
+        let underline: StyleData
+    }
+    
+    private struct ListInsertData {
+        let text: String
+        let index: Int
+        let stringIndex: String.Index
+    }
+    
+    private struct TagRemoveData {
+        let range: NSRange
+    }
+    
+    private struct TagAndContentRemoveData {
+        let range: NSRange
+    }
+    
+    // MARK: - NSAttributedString - Public
+    
+    public static func nsAttributedStringFromHtml(_ html: String, styles: Styles) throws -> NSAttributedString {
+        let attributedString = NSMutableAttributedString(string: html)
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineSpacing = 3
```

Comment:
What do you think about propagating this value through `Styles` as well?

Reply:
Done! One thing that's still a bit awkward here is that SwiftUI adds line spacing as a modifier, outside of the utility. I couldn't find any line spacing key while making AttributedString. I guess we could consider making a `WKHtmlText` component or something, just for SwiftUI, with line spacing applied within so that we don't have to remember to add this modifier every time. Let me know what you think.

Reply:
Yeah a component that itself encapsulates those custom modifiers makes sense in the future to me too.